rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+      - [Current monorepo](#current-monorepo)
+        - [Pros](#pros-3)
+        - [Cons](#cons-3)
+      - [Separate monorepo](#separate-monorepo)
+        - [Pros](#pros-4)
+        - [Cons](#cons-4)
+    - [Package scope](#package-scope)
+      - [@fluentui scope](#fluentui-scope)
+        - [Pros](#pros-5)
+        - [Cons](#cons-5)
+      - [New package scope](#new-package-scope)
+        - [Pros](#pros-6)
+        - [Cons](#cons-6)
+    - [How users should consume components](#how-users-should-consume-components)
+      - [Suite packages](#suite-packages)
+        - [Pros](#pros-7)
+        - [Cons](#cons-7)
+      - [Individual packages](#individual-packages)
+        - [Pros](#pros-8)
+        - [Cons](#cons-8)
+    - [How to depend on core components](#how-to-depend-on-core-components)
+      - [Depend directly on components packages](#depend-directly-on-components-packages)
+        - [Pros](#pros-9)
+        - [Cons](#cons-9)
+      - [Depend on suite package is peer dependency](#depend-on-suite-package-is-peer-dependency)
+      - [Pros](#pros-10)
+      - [Cons](#cons-10)
+    - [Documentation](#documentation)
+      - [Current docsite](#current-docsite)
+        - [Pros](#pros-11)
+        - [Cons](#cons-11)
+      - [New docsite](#new-docsite)
+        - [Pros](#pros-12)
+        - [Cons](#cons-12)
+    - [Parnter ownership and collaboration](#parnter-ownership-and-collaboration)
+      - [GitHub](#github)
+        - [Pros](#pros-13)
+        - [Cons](#cons-13)
+      - [ADO mirroring](#ado-mirroring)
+        - [Pros](#pros-14)
+        - [Cons](#cons-14)
+      - [Both](#both)
+        - [Pros](#pros-15)
+        - [Cons](#cons-15)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# RFC: Support for components in the Fluent UI umbrella that are not core
+
+[@ling1726](https://github.com/ling1726) [@layershifter](https://github.com/layershifter)
+
+## Summary
+
+@fluentui/react-components has matured, with many flexible primitives that are also highly customizable. These
+primitive components generally don't reuse each other and are considered the 'building blocks' of a design system.
+Future design iterations might introduce components that are either:
+
+- Very situational - i.e. applied to specific scenarios only
+- Composed of multiple primitive components
+- Specializations of primitive components
+
+We need to formalise how we will deliver these components not only as first party packages but also second party
+packages with our partners who might need more situational components powered by V9. This RFC refers to these kinds
+of components as `non-core`.
+
+## Problem statement
+
+We create components that will not be shipped as a part of our 'core' components under the `@fluentui/react-components`
+library but other packages. We will have to finalize the following requirements:
+
+- Should non-core components be able to deviate from V9 core concepts?
```

Comment:
I think yes. they will be owned by other teams, we should give them trust to make the right decisions. they might need to opt out of compliance checks.

Reply:
define 'deviate'...are we talking about slots, or file structure, or styling or what. 

I think there is a good question here as to 'how much can a component deviate from the v9 concepts while still being at all related to v9'

If I write a quick (props)=> <div>{props.children}</div>

is this a fluent ui component as long as it matches a design somewhere in a figma?

Reply:
I don't think we should try to be too restrictive here for non-core components. For non-core, I'd even propose that no one needs our permission to call something part of the Fluent ecosystem.

I do think, however, there are some guidelines. 
- It should follow the Fluent design guidelines
- It should depend on the Fluent/v9 core SDK, which means respecting design tokens
- It should not re-invent existing primitives, but it may not depend on any primitives